discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -65,47 +63,179 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-class PrometheusType:
-    HISTOGRAM = "histogram"
-    GAUGE = "gauge"
+METRIC_PREFIX = "audius_dn"
+
+
+class PrometheusMetricNames:
+    """
+    Attempt to group metrics with high_level prefixes like:
+    * `flask_`
+    * `celery_task_`
+
+    Antepenultimate Suffixes:
```

Comment:
these labels like antepenultimate, penultimate seem complicated and i had to look them up and i'm not sure it really helped me? i feel like just having the list of suffixes is probably fine

Reply:
Yeah, I also had to look up "antepenultimate" and it turns out there are proper words all the way up to 5th-from-last. 🤓 

The ordering here is most important, which is why these labels are included. Having the same ordering of suffixes makes it easier to reason about as we introduce more metrics.

This is essentially the same doc that's in Notion that matches the same structure as the [official docs](https://prometheus.io/docs/practices/naming/). I'll merge this in today as is, but we can always follow up with a PR specific for this docstring.

Thanks for the review!